las_geoh5/import_las.py

```diff
@@ -0,0 +1,192 @@
+        file = lasio.read(survey, mnemonic_case="preserve")
+        try:
+            surveys = np.c_[file["DEPT"], file["DIP"], file["AZIM"]]
+            if len(drillhole.surveys) == 1:  # type: ignore
+                drillhole.surveys = surveys  # type: ignore
+        except KeyError:
+            warnings.warn(
+                "Attempted survey import failed because data read from "
+                ".las file did not contain the expected 3 curves 'DEPT'"
+                ", 'DIP', 'AZIM'."
+            )
+    else:
+        surveys = np.genfromtxt(survey, delimiter=",", skip_header=0)
+        if surveys.shape[1] == 3:  # type: ignore
+            if len(drillhole.surveys) == 1:  # type: ignore
+                drillhole.surveys = surveys  # type: ignore
+        else:
+            warnings.warn(
+                "Attempted survey import failed because data read from "
+                "comma separated file did not contain the expected 3 "
+                "columns of depth/dip/azimuth."
+            )
+
+    return drillhole
+
+
+def create_or_append_drillhole(
+    workspace: Workspace,
+    lasfile: lasio.LASFile,
+    drillhole_group: DrillholeGroup,
+    property_group: str | None = None,
+):
+    """
+    Create a drillhole or append data to drillhole if it exists in workspace.
+
+    :param workspace: Project workspace.
+    :param lasfile: Las file object.
+    :param drillhole_group: Drillhole group container.
+    :param property_group: Property group name.
+
+    :return: Created or augmented drillhole.
+    """
+
+    name = lasfile.well["WELL"].value
+    collar = [
+        lasfile.well.get("X", None),
+        lasfile.well.get("Y", None),
+        lasfile.well.get("ELEV", None),
+    ]
+    if all(k is not None for k in collar):
+        collar = [k.value for k in collar]
+    else:
+        collar = None  # type: ignore
+
+    drillhole = drillhole_group.get_entity(name)
+    drillhole = drillhole[0] if drillhole else None  # type: ignore
+    if drillhole is None:
+        kwargs = {"name": name}
+        kwargs["parent"] = drillhole_group
+        if collar:
+            kwargs["collar"] = collar
+
+        drillhole = Drillhole.create(workspace, **kwargs)
+
+    elif not np.allclose(collar, drillhole.collar.tolist()):  # type: ignore
+        kwargs = {"name": find_copy_name(workspace, drillhole.name)}  # type: ignore
+        kwargs["parent"] = drillhole_group
+        if collar:
+            kwargs["collar"] = collar
+
+        drillhole = Drillhole.create(workspace, **kwargs)
+
+    pg_type = "Interval table" if "TO" in lasfile.curves else "Depth table"
+    property_group = drillhole.find_or_create_property_group(  # type: ignore
+        name=property_group, property_group_type=pg_type, association="DEPTH"
+    )
+    for curve in [k for k in lasfile.curves if k.mnemonic not in ["DEPT", "TO"]]:
+        name = curve.mnemonic
+        data = drillhole.get_data(name)  # type: ignore
+
+        if not data:

```

Comment:
Reduce indentation with `continue` on the reverse logic. 
On that note, is there a case where data would come in twice?

Reply:
Done.  lasio enforces unique curve names, so I don't think so (if i understood you correctly).